packages/e2e/test/web-extension/dapp-sdk/features/singleDelegation.ts

```diff
@@ -0,0 +1,62 @@
+import { Cardano } from '@cardano-sdk/core';
+import { ObservableWallet } from '@cardano-sdk/wallet';
+import { buildAndSignTx } from '../utils';
+import { combineLatest, take, tap } from 'rxjs';
+import { toSerializableObject } from '@cardano-sdk/util';
+
+export const singleDelegation = ({
+  connectedWallet,
+  logger
+}: {
+  logger: typeof console;
+  connectedWallet: ObservableWallet;
+}) => {
+  const delegateElement = document.querySelector('#info-delegate-assets')!;
+  const delegateAssetsElement = document.querySelector('#info-delegate-asset-id')!;
+
+  combineLatest([
+    connectedWallet.addresses$,
+    connectedWallet.balance.utxo.available$,
+    connectedWallet.delegation.rewardAccounts$
+  ])
+    .pipe(
+      take(1),
+      tap(async ([addresses, availableBalance, rewardAccounts]) => {
+        if (!connectedWallet) {
+          return logger.warn('Please, connect the wallet first');
+        }
```

Comment:
We can trust the types since this is not nullable, and we're making this assertion in dapp-sdk.ts already. Same for undelegation and sendSeveralAssets.
```suggestion
```

Reply:
it's addressed here: ac7ca1e3962ca8240a8e701b545712cc53127eaa